flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/factories/datagen/SequenceGeneratorVisitor.java

```diff
@@ -0,0 +1,177 @@
+
+		ConfigOptions.OptionBuilder startKey = key(startKeyStr);
+		ConfigOptions.OptionBuilder endKey = key(endKeyStr);
+
+		config.getOptional(startKey.stringType().noDefaultValue()).orElseThrow(
+			() -> new ValidationException(
+				"Could not find required property '" + startKeyStr + "' for sequence generator."));
+		config.getOptional(endKey.stringType().noDefaultValue()).orElseThrow(
+			() -> new ValidationException(
+				"Could not find required property '" + endKeyStr + "' for sequence generator."));
+
+		this.intStart = startKey.intType().noDefaultValue();
+		this.intEnd = endKey.intType().noDefaultValue();
+		this.longStart = startKey.longType().noDefaultValue();
+		this.longEnd = endKey.longType().noDefaultValue();
+	}
+
+	@Override
+	public DataGeneratorContainer visit(BooleanType booleanType) {
+		return DataGeneratorContainer.of(RandomGenerator.booleanGenerator());
+	}
+
+	@Override
+	public DataGeneratorContainer visit(CharType booleanType) {
+		return DataGeneratorContainer.of(
+			getSequenceStringGenerator(
+				config.get(longStart), config.get(longEnd)),
+			longStart, longEnd);
+	}
+
+	@Override
+	public DataGeneratorContainer visit(VarCharType booleanType) {
+		return DataGeneratorContainer.of(
+			getSequenceStringGenerator(
+				config.get(longStart), config.get(longEnd)),
+			longStart, longEnd);
+	}
+
+	@Override
+	public DataGeneratorContainer visit(TinyIntType booleanType) {
+		return DataGeneratorContainer.of(
+			SequenceGenerator.byteGenerator(
+				config.get(intStart).byteValue(),
+				config.get(intEnd).byteValue()),
+			intStart, intEnd);
+	}
+
+	@Override
+	public DataGeneratorContainer visit(SmallIntType booleanType) {
+		return DataGeneratorContainer.of(
+			SequenceGenerator.shortGenerator(
+				config.get(intStart).shortValue(),
+				config.get(intEnd).shortValue()),
+			intStart, intEnd);
+	}
+
+	@Override
+	public DataGeneratorContainer visit(IntType integerType) {
+		return DataGeneratorContainer.of(
+			SequenceGenerator.intGenerator(
+				config.get(intStart), config.get(intEnd)),
+			intStart, intEnd);
+	}
+
+	@Override
+	public DataGeneratorContainer visit(BigIntType bigIntType) {
+		return DataGeneratorContainer.of(
+			SequenceGenerator.longGenerator(
+				config.get(longStart), config.get(longEnd)),
+			longStart, longEnd);
+	}
+
+	@Override
+	public DataGeneratorContainer visit(FloatType floatType) {
+		return DataGeneratorContainer.of(
+			SequenceGenerator.floatGenerator(
+				config.get(intStart).shortValue(),
+				config.get(intEnd).shortValue()),
+			intStart, intEnd);
+	}
+
+	@Override
+	public DataGeneratorContainer visit(DoubleType doubleType) {
+		return DataGeneratorContainer.of(
+			SequenceGenerator.doubleGenerator(
+				config.get(intStart), config.get(intEnd)),
+			intStart, intEnd);
+	}
+
+	@Override
+	public DataGeneratorContainer visit(DecimalType decimalType) {
+		return DataGeneratorContainer.of(
+			SequenceGenerator.bigDecimalGenerator(
```

Comment:
Looks like the structures of Decimal, Times are wrong, you can take a look to the comments of `RowData`.